runtime/compiler/control/J9Options.cpp

```diff
@@ -1599,6 +1602,26 @@ void J9::Options::preProcessJniAccelerator(J9JavaVM *vm)
       }
    }
 
+uint64_t getCodeCacheMaxPercentageOfAvailableMemory(J9JavaVM *vm)
+   {
+   PORT_ACCESS_FROM_JAVAVM(vm);
+   OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+
+   uint64_t codeCacheTotalPercentage = CODECACHE_DEFAULT_RAMPERCENTAGE;
+   char *xxccPercentOption = J9::Options::_externalOptionStrings[J9::ExternalOptions::XXcodecachetotalRAMPercentage];
+   int32_t XXcodeCacheTotalPercentArg = FIND_ARG_IN_VMARGS(EXACT_MEMORY_MATCH, xxccPercentOption, 0);
+   if (XXcodeCacheTotalPercentArg >= 0)
+      {
+      IDATA returnCode = GET_MEMORY_VALUE(XXcodeCacheTotalPercentArg, xxccPercentOption, codeCacheTotalPercentage);
```

Comment:
GET_MEMORY_VALUE may work, but it's typically used for reading memory values with a suffix (like 512M).
I see that there is a GET_PERCENT_VALUE macro, which can be used instead.

Reply:
GET_PERCENT_VALUE takes a value between 0 and 1 and returns 0 to anything outside that range. I found GET_DOUBLE_VALUE to work better.